Add devise gem and views

AyoubM490 opened the pull request:
In this commit, I added the devise gem and setup the devise views

AyoubM490 committed:
Add devise gem and views

AyoubM490 committed:
fix rubocop linter errors

keroloussamy commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -0,0 +1,29 @@
+<h2>Sign up</h2>
+
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %>
+    <% if @minimum_password_length %>
+    <em>(<%= @minimum_password_length %> characters minimum)</em>
+    <% end %><br />
+    <%= f.password_field :password, autocomplete: "new-password" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
```
- [x] Kindly when I try to register as a new user, it displays an error to add the Name but there's no input for this, Kindly fix this issue.

![image](https://user-images.githubusercontent.com/35535130/179854214-6b1cc27f-7371-4e44-8eca-0c1eea0dfc35.png)

keroloussamy commented on app/models/user.rb:
```diff
@@ -1,4 +1,9 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
- [x] Kindly add `:confirmable` to the devise options to let your application ask for confirmation for each email. As required by the requirements.

keroloussamy reviewed (changes requested):
Hi @AyoubMs, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._

![](https://camo.githubusercontent.com/6ac905aa3a075ef45de55758c20c6e6efb1dfdbbb0d9fd03d4718d26c5c96cde/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f4a713832345239334a734c775a436169534c2f67697068792e676966)


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

AyoubM490 committed:
- add name text_field
- add :confirmable option to the devise options to let the app ask for
confirmation for each email

AyoubM490 committed:
fix rubocop linter errors

Kingstalux commented on app/models/user.rb:
```diff
@@ -2,16 +2,16 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+         :recoverable, :rememberable, :validatable, :confirmable
```
- [x] Good job adding `confirmable` but now when I register a new user I get the following error. To resolve this, kindly uncomment lines 25 to 28 in the devise migration file 👍 
- Error:
![Screenshot 2022-07-20 082138](https://user-images.githubusercontent.com/85113567/179922397-a8cfa8fc-520e-4ab9-95bb-873d1149ee28.png)

Kingstalux commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -1,29 +1,27 @@
 <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
   <%= render "devise/shared/error_messages", resource: resource %>
-
+  <div class="field">
+    <%= f.label :name %><br />
+    <%= f.text_field :name, autofocus: true, autocomplete: "name" %>
+  </div>
```
- [x] (OPTIONAL) Great job adding the name field 👍 It would be great if you could add fields for the other user inputs like bio and photo 💯 
- [x] (OPTIONAL) Also, it would be great if you could configure your routes to take a user to the signup page if that user hasn't signed up yet 😄

Kingstalux reviewed (changes requested):
Hi @AyoubMs, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![175083599-879bf673-fb93-406f-939c-de7e586bf6d9](https://user-images.githubusercontent.com/85113567/179923139-9819ada4-4db2-4c5e-a858-8344d6e4a841.gif)

 ## Highlights 👏 
✅ Good commit history 👍 
✅ Linters are passing 👌 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

AyoubM490 committed:
Fix confirmation
add bio and photo
configuring routes to take user to the signup page if that user hasn't signed
up yet

AyoubM490 committed:
fix rubocop linter errors

sja-thedude reviewed (changes requested):
## Required Changes ♻️

Hi @AyoubMs, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![200w](https://user-images.githubusercontent.com/70269507/159484303-3187dacc-4a82-4263-bb1a-8d72417ce1d9.gif)

 ## Highlights
✔️ Good job so far

 ## Required Changes ♻️
- [x] While running rails db:migrate, I am encountering an error because of which I couldn't check whether the app is running perfectly. Kindly fix the error. 
<img width="1022" alt="Screenshot 2022-07-20 at 3 00 42 PM" src="https://user-images.githubusercontent.com/70269507/179950102-e3516f42-8c47-401f-97bb-e9ce122c749f.png">

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

AyoubM490 committed:
add email column

medaminedev66 reviewed (approved):
Hi @AyoubMs 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💯 
Congratulations! 🎉 

### Highlights
- Your rails app works as expected and runs without error.
- The user can sign up without encountering any issues.
- The user can log in without encountering any issues.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
